misc.c

```diff
@@ -1301,7 +1301,17 @@ vdollar_percent_expand(int *parseerror, int dollar, int percent,
 			}
 			var = xmalloc(len + 1);
 			(void)strlcpy(var, string, len + 1);
-			if ((val = getenv(var)) == NULL) {
+			char RANDOM[7];
+			val = NULL;
+			if (strcmp(var, "RANDOM") == 0) {
+				/*
+				 * Generate a random 5-digit number to support the
+				 * special ${RANDOM} environment variable.
+				 */
+				sprintf(RANDOM, "%d", arc4random_uniform(1e6));
```

Comment:
Using arc4random here adds an indirect dependency on OpenSSL's libcrypto.  We put some effort into avoiding that dependency for binaries like scp, sftp and sftp-server which is why many of the Github Actions tests fail.

This could be avoided by moving this function into one of the files only linked into ssh and sshd (but see above about whether or not this approach is the right way to solve this, leaving aside the implementation).

Reply:
I replied in the bugzilla bug with an attempt to clarify the reasons why this feature would be useful for me, I understand this is still a discussion.

Thanks for helping me understand why some of those github action tests fail. I'm not very familiar with the codebase so if you could point me into what you think would be the right file where to move this function that would be very helpful.